kedro-datasets/test_requirements.txt

```diff
@@ -7,7 +7,7 @@ black~=22.0
 compress-pickle[lz4]~=1.2.0
 coverage[toml]
 dask[complete]~=2021.10  # pinned by Snyk to avoid a vulnerability
-delta-spark~=1.2.1  # 1.2.0 has a bug that breaks some of our tests: https://github.com/delta-io/delta/issues/1070
+delta-spark
```

Comment:
Yeah, sure. But just a reminder here, those pins arent quite being respected. The dependency resolver is just finding a solution that "works", like Ive commented earlier. (E.g: pandas~=1.3, but its 1.5.3 thats being used). 
Honestly, why were not using `pip install -e ".[all]"` ? (I may be missing the bigger picture)

Reply:
You're absolutely right. IMO we don't necessarily have the most optimal testings and development flow for the datasets. We're improving our dev setup, but it's a slow process. I guess for the time being it would just be nice to keep it consistent with how other dependencies are pinned.